### PR TITLE
Validate payment number string before multiplying

### PR DESCRIPTION
**Pull request by yuliaveres**

### CODE REVIEW

File: Token/Controllers/PaymentController.swift [hunk 1]
```diff
@@ -135,6 +135,9 @@ extension PaymentController: UITextFieldDelegate {
             if let currencyString = (textField.text as NSString?)?.replacingCharacters(in: range, with: string) {
                 if let currencyValue = self.inputNumberFormatter.number(from: currencyString) {
                     let ether = EthereumConverter.localFiatToEther(forFiat: currencyValue)
+                    
+                    guard ether != NSDecimalNumber.notANumber else { return }
```
Comment (Elland) [inline Token/Controllers/PaymentController.swift]: Is the `NSDecimalNumber` necessary or can we just say `ether != .notANumber`? Also I find the double negative here a bit confusing to follow, but I guess no easy way around that.

Comment (yuliaveres) [inline Token/Controllers/PaymentController.swift]: checking

File: Token/Controllers/PaymentController.swift [hunk 2]
```diff
@@ -136,10 +136,10 @@ extension PaymentController: UITextFieldDelegate {
                 if let currencyValue = self.inputNumberFormatter.number(from: currencyString) {
                     let ether = EthereumConverter.localFiatToEther(forFiat: currencyValue)
                     
-                    guard ether != NSDecimalNumber.notANumber else { return }
-                    
-                    self.valueInWei = ether.multiplying(byPowerOf10: EthereumConverter.weisToEtherPowerOf10Constant)
-                    self.etherAmountLabel.text = EthereumConverter.ethereumValueString(forEther: ether)
+                    if ether != .notANumber {
```
Comment (Elland) [inline Token/Controllers/PaymentController.swift]: Maybe just an `if ether == .notANumber { return }` instead? Then there's no double negative.

Comment (yuliaveres) [inline Token/Controllers/PaymentController.swift]: No return inside defer 😢

File: Toshi/Controllers/PaymentController.swift [hunk 3]
```diff
@@ -136,7 +136,7 @@ extension PaymentController: UITextFieldDelegate {
                 if let currencyValue = self.inputNumberFormatter.number(from: currencyString) {
                     let ether = EthereumConverter.localFiatToEther(forFiat: currencyValue)
                     
-                    if ether != .notANumber {
+                    if ether.isANumber == true {
```
Comment (Elland) [inline Toshi/Controllers/PaymentController.swift]: if `isANumber` is a bool, do we need to check against `true`? Just realised we were doing that.

File: Toshi/Controllers/PaymentController.swift [hunk 4]
```diff
@@ -136,7 +136,7 @@ extension PaymentController: UITextFieldDelegate {
                 if let currencyValue = self.inputNumberFormatter.number(from: currencyString) {
                     let ether = EthereumConverter.localFiatToEther(forFiat: currencyValue)
                     
-                    if ether != .notANumber {
+                    if ether.isANumber {
```
Comment (Elland) [inline Toshi/Controllers/PaymentController.swift]: 💪🏽

File: Toshi/Extensions/NSDecimalNumber+Additions.swift [hunk 5]
```diff
@@ -28,6 +28,10 @@ public extension NSDecimalNumber {
     public var toHexString: String {
         return "0x\(BaseConverter.decToHex(self.toDecimalString).lowercased())"
     }
+    
+    public var isANumber: Bool {
+        return self != .notANumber
```
Comment (Elland) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: Same here. `return !.notANumber`. 👍

Comment (yuliaveres) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: come on
what does that mean?

Comment (yuliaveres) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: and you can't do this
`Type 'Bool' has no member 'notANumber'`

Comment (Elland) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: Simpler than `return self != self.notANumber`, no? That feels super redundant to me. Even with the implicit `self` there.

Comment (Elland) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: And it being hard to read either way, I guess this is not avoidable, it's the reason we extended the class anyway? 😅

Comment (yuliaveres) [inline Toshi/Extensions/NSDecimalNumber+Additions.swift]: i think we achieved syntactical improvements :) i am fine with this, not sure about other people

**Pull request merged**